src/main/java/jenkins/advancedqueue/jobinclusion/strategy/FolderBasedJobInclusionStrategy.java

```diff
@@ -62,9 +71,33 @@ public ListBoxModel getListFolderItems() {
 
 	};
 
+	@Restricted(NoExternalUse.class)
+	static public class JobPattern {
+		private String jobPattern;
+
+		@DataBoundConstructor
+		public JobPattern(String jobPattern) {
+			this.jobPattern = jobPattern;
+		}
+
+	}
+
 	private String folderName;
 
+	private boolean useJobFilter = false;
```

Comment:
Actually there is no need to keep and persist this flag. `jobPattern  != null` should be enough